dask/dataframe/tests/test_indexing.py

```diff
@@ -410,6 +410,25 @@ def test_to_series():
     assert_eq(df.index.to_series(), ddf.index.to_series())
 
 
+def test_to_frame():
+
+    # Test for time index
+    df = pd.DataFrame({'A': np.random.randn(100)},
+                      index=pd.date_range('2011-01-01', freq='H', periods=100))
+    ddf = dd.from_pandas(df, 10)
+
+    assert_eq(df.index.to_frame(), ddf.index.to_frame())
+    assert_eq(df.index.to_frame(name='foo'), ddf.index.to_frame(name='foo'))
```

Comment:
Using the `name=` keyword here will cause this test to fail when testing with pandas versions < 0.24.0. Perhaps we can have two separate tests, `test_to_frame`, which doesn't set the `name=` keyword, and `test_to_frame_name`, which does set `name=`. `@pytest.mark.skip` can be used to skip `test_to_frame_name` when the pandas version is < 0.24.0